### DISCUSSION TITLE
Bills - Pay Early Every Month (Workaround)

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [firefly-iii/firefly-iii#7720] [General]
I have the following scenario. This may not match directly with what another user deals with but hopefully it gives some ideas.

## Setup
My rent is due on the 1st of every month. You'd think this would mean I just tell Firefly I pay on the 1st and set the repeat to monthly, but this doesn't actually work for my scenario. I get paid bi-weekly, and I typically pay my rent with my last paycheck every month. So to configure this, I did the following:

Date: (the date of last month's last paycheck)
Repeats: Weekly
Skip: 3

This means that every fourth week, it expects me to pay my rent bill. This doesn't reflect when it is actually due, but makes sure that when I report my rent payment to Firefly it applies it to the correct period. If I were to set it up monthly on the first, then when I pay my rent the Thursday before the end of the month, it applies to the *last* period, meaning that from the perspective of Firefly I actually made my rent payment 3 weeks late.

## Drawbacks
I've noticed the following drawbacks of this solution:
1. Because I'm paid bi-weekly, two months out of the year (typically June and December) have three paychecks, which means bi-annually my pay calculation for a bill using this method will be off-by-one. To fix this an update to the date field is necessary, but I feel that's relatively minor.
2. We are lying to Firefly about when the bill is actually due to get it to work with how we pay the bill.

## How To Fix The Bodge
Add a field which allows us to set the period independently from when the bill is due. For instance, I might set the bill to repeat monthly on the 1st but with a period offset of -7 days. This means that any transactions dated 7 days before the 1st of July get applied to the month of July, not the month of June. I've not reviewed the code for this section but an offset like this will likely require the least amount of development work.

## Final Thoughts
Please feel free to leave ideas and comments and critiques. I'm curious to see what other workaround have worked for you or if there are any ways to make this work a bit better for my scenario.

**Comment by JC5** [comment]
Why are you limiting yourself to your biweekly salary schedule?

What I do is transfer my income to my savings account, and pay myself from that account every month.

Then, it's suddenly a lot easier to manage this

**Reply by TheBrambleShark** [reply]
I live paycheck-to-paycheck, so I tend to use Firefly in a manner that reflects my life.
